raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -193,8 +193,9 @@ private void updateFollowerHighWatermark(
     ) {
         highWatermarkOpt.ifPresent(highWatermark -> {
             long newHighWatermark = Math.min(endOffset().offset, highWatermark);
-            state.updateHighWatermark(OptionalLong.of(newHighWatermark));
-            updateHighWatermark(state, currentTimeMs);
+            if (state.updateHighWatermark(OptionalLong.of(newHighWatermark))) {
```

Comment:
Just to clarify this is not a correctness bugfix, but just to optimize away unnecessary purgatory access right?

Reply:
Yeah, right. Also to avoid the log spam when the high watermark doesn't actually increment.